components/drivers/i2c/i2c_core.c

```diff
@@ -118,16 +118,16 @@ rt_size_t rt_i2c_master_send(struct rt_i2c_bus_device *bus,
 
     ret = rt_i2c_transfer(bus, &msg, 1);
 
-    return (ret > 0) ? count : ret;
+    return ret;
```

Comment:
Not back-compatible. It SHOULD return byte count value if sent an I2C message successfully.